src/solve/test.rs

```diff
@@ -536,21 +536,47 @@ fn elaborate_normalize() {
     test! {
         program {
             trait Eq { }
+            struct i32 { }
 
             trait Item where <Self as Item>::Out: Eq {
                 type Out;
             }
+
+            impl Eq for i32 { }
+            impl Item for i32 {
+                type Out = i32;
+            }
         }
 
         goal {
             forall<T, U> {
-                if (T: Item<Out = U>) {
+                if (WellFormed(T: Item), T: Item<Out = U>) {
```

Comment:
why is this `WF(T: Item)` and not `WF(T: Item<Out = U>)`? Because the former suffices?